utils/fixed_decimal/src/decimal.rs

```diff
@@ -1282,21 +1302,93 @@ impl FixedDecimal {
     /// assert_eq!("4", dec.to_string());
     /// ```
     pub fn half_trunc(&mut self, position: i16) {
-        let digit_after_position = self.digit_at_next_position(position);
-        let should_expand = match digit_after_position.cmp(&5) {
-            Ordering::Less => false,
-            Ordering::Greater => true,
-            Ordering::Equal =>
-            // NOTE: `digit_after_position` equals 5, this means, position does not equal to `i16::MIN`.
-            {
-                self.nonzero_magnitude_end() < position - 1
+        self.half_trunc_to_increment(position, RoundingIncrement::MultiplesOf1);
+    }
+
+    /// Half Truncates the number on the right to a particular position and rounding increment,
+    /// deleting digits if necessary.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use fixed_decimal::{FixedDecimal, RoundingIncrement};
+    /// # use std::str::FromStr;
+    ///
+    /// let mut dec = FixedDecimal::from_str("-3.5").unwrap();
+    /// dec.half_trunc_to_increment(0, RoundingIncrement::MultiplesOf2);
+    /// assert_eq!("-4", dec.to_string());
+    /// let mut dec = FixedDecimal::from_str("7.57").unwrap();
+    /// dec.half_trunc_to_increment(-1, RoundingIncrement::MultiplesOf5);
+    /// assert_eq!("7.5", dec.to_string());
+    /// let mut dec = FixedDecimal::from_str("5.45").unwrap();
+    /// dec.half_trunc_to_increment(-2, RoundingIncrement::MultiplesOf25);
+    /// assert_eq!("5.50", dec.to_string());
+    /// let mut dec = FixedDecimal::from_str("9.99").unwrap();
+    /// dec.half_trunc_to_increment(-1, RoundingIncrement::MultiplesOf25);
+    /// assert_eq!("10.0", dec.to_string());
+    /// let mut dec = FixedDecimal::from_str("9.99").unwrap();
+    /// dec.half_trunc_to_increment(-2, RoundingIncrement::MultiplesOf2);
+    /// assert_eq!("9.98", dec.to_string());
+    /// ```
+    pub fn half_trunc_to_increment(&mut self, position: i16, increment: RoundingIncrement) {
+        fn next_digits_exceed_five(number: &mut FixedDecimal, position: i16) -> bool {
```

Comment:
Suggestion: it may be helpful to add the following private function on FixedDecimal and use it wherever you need it:

```rust
enum HalfAtMagnitude {
    LessThanHalf,
    ExactlyHalf,
    GreaterThanHalf,
}

fn half_at_magnitude(&self, magnitude: i16) -> HalfAtMagnitude {
    match (self.nonzero_magnitude_end().cmp(magnitude)), self.digit_at(magnitude).cmp(5)) {
        (Ordering::Less, Ordering::Less) => LessThanHalf, // 0.1x
        (Ordering::Less, _) => GreaterThanHalf, // 0.5x, 0.9x
        (Ordering::Equal, Ordering::Less) => LessThanHalf, // 0.1
        (Ordering::Equal, Ordering::Equal) => ExactlyHalf, // 0.5
        (Ordering::Equal, Ordering::Greater) => GreaterThanHalf, // 0.9
        (Ordering::Greater, _) => debug_assert!(false, "don't call this function in this case"),
    }
}
```

You can use this function as part of checking for boundaries at 0.5, 2.5, 7.5, 12.5, 37.5, 62.5, 87.5

Reply:
Had to slightly modify the method since it was more useful to use the digits at the right of `magnitude`, and opted for using `Ordering` because I think it carries the same semantics as `HalfAtMagnitude`.